src/common/Common.ts

```diff
@@ -91,39 +92,41 @@ export class CommandLineUtils {
      */
     public static platformFlagIsAndroid(input: string): boolean {
         if (input) {
-            return input.toLowerCase() === CommandLineUtils.ANDROID_FLAG;
+            return input.trim().toLowerCase() === CommandLineUtils.ANDROID_FLAG;
+        }
+        return false;
+    }
+
+    /**
+     * Checks to see if a flag is targeting Desktop.
+     * @param input The input flag.
+     * @returns True if flag is targeting Desktop.
+     */
+    public static platformFlagIsDesktop(input: string): boolean {
+        if (input) {
+            return input.trim().toLowerCase() === CommandLineUtils.DESKTOP_FLAG;
         }
         return false;
     }
 
     /**
      * Checks to see if a platform flag has a valid value.
      * @param platformFlag The platform flag.
-     * @returns True if flag is valid (i.e either targeting iOS or Android).
+     * @param includeDesktop Indicates whether Desktop is allowed as a target platform. Defaults to false.
+     * @returns True if flag is valid.
      */
-    public static platformFlagIsValid(platformFlag: string) {
+    public static platformFlagIsValid(
```

Comment:
The changes to this file are my attempt at using a simple way (with minimal coding) of adding support for having commands that might want to target Desktop as a platform. With the exception of  the Preview command, currently most of our commands don't need to target desktop as a platform and that is why `includeDesktop` defaults to `false`. When validating its parameters, the Preview command will call `platformFlagIsValid` and pass `includeDesktop=true`.

This is a simple way of having the Preview command handle all 3 platform, instead of having code in our vscode extension to detect Desktop as a special case and handle it by itself, but then let the Preview command handle Android/iOS platforms. With the simple changes in this file, the code for our vscode extension will be greatly simplified as it will now just pass the parameters for any platform to the Preview command and let it do its thing.

Reply:
I think that's fine for the general purposes of writing a consolidation, as that's likely something that should be done in the longer term. We should work with the platform dev team to decide what a final home for that logic should look like, that can satisfy both projects. But I agree that at a minimum this is a good exercise to prove out how that should work.